<launch>
  <arg name="port" default="/dev/ttyUSB0"/>
  <arg name="vacuum_thres" default="642"/>
  <include file="$(find arm_operation)/launch/change_tool_service.launch"/>
  <node pkg="arm_operation" type="agent_server_node" name="agent_server_node" output="screen" clear_params="true">
    <rosparam param="tool_length">[0.15, 0.084, 0.05]</rosparam>
    <rosparam param="service_name">["/change_tool_service/change_tool_service",
                                    "/abb/GetCartesian",
                                    "/abb/SetCartesian",
                                    "/abb/GetJoints",
                                    "/abb/SetJoints",
                                    "/abb/SetZone",
                                    "/abb/SetSpeed",
                                    "/vacuum_pump_control_node/vacuum_control"]</rosparam>
    <rosparam param="home_joints">[-0.37856191396713257, 
                                   0.2209586799144745, 
                                   0.08604472875595093, 
                                   -0.018849557265639305, 
                                   1.2630947828292847, 
                                   -0.4089306592941284]</rosparam>
    <rosparam param="place_joints">[0.18395769596099854, 
                                    0.14468780159950256, 
                                    0.17662732303142548, 
                                    -0.016406094655394554, 
                                    1.2583823204040527, 
                                    0.15271630883216858]</rosparam>
  </node>
  <node pkg="vacuum_conveyor_control" type="vacuum_pump_control" name="vacuum_pump_control_node" output="screen">
    <param name="port" value="$(arg port)"/>
    <param name="vacuum_thres" value="$(arg vacuum_thres)"/>
  </node>
</launch>
